src/Common/src/System/Security/Cryptography/DerSequenceReader.cs

```diff
@@ -110,6 +110,78 @@ internal byte[] ReadNextEncodedValue()
             return encodedValue;
         }
 
+        /// <summary>
+        /// This method returns the next value encoded (including tag and length), as
+        /// a byte[][3] to be fed to DerEncoder when needed.
+        /// </summary>
+        internal byte[][] ReadAndSplitNextEncodedValue()
+        {
+            int lengthLength;
+            int contentLength = ScanContentLength(_data, _position + 1, out lengthLength);
+
+            byte[] encodedLength = new byte[lengthLength];
+
+            byte tag = _data[_position++];
```

Comment:
Personally, I hate side-effect expressions buried in other expressions, and would move the increment operation to the next line.